---
uint16 way
string lan